/*
* Copyright (c) 2016 Cisco and/or its affiliates,
*                    MaxLinear, Inc. ("MaxLinear"), and
*                    Cable Television Laboratories, Inc. ("CableLabs")
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at:
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
syntax = "proto2";

message t_l2tpVspAvp {
    required uint32 vendorId           = 1;
    required uint32 attrType           = 2;
    required uint32 attrValLen         = 3;
    required string attrValBuf         = 4;
    }

message t_l2tpVspAvpMsg {
    enum t_Operation {
        VSP_AVP_OP_INFO   = 1;
        VSP_AVP_OP_UPDATE = 2;
    }

	enum t_ResponseCode {
    	VSP_AVP_STATUS_NA      = 1;
    	VSP_AVP_STATUS_SUCCESS = 2;
	    VSP_AVP_STATUS_FAILURE = 3;
	}

    required t_Operation oper          = 1;
    required t_ResponseCode rspCode    = 2;
    repeated t_l2tpVspAvp VspAvp       = 3;
}
